mapper/orm/src/main/java/org/hibernate/search/mapper/orm/search/query/impl/HibernateOrmSearchQuery.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.orm.search.query.impl;
+
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import java.util.Optional;
+import javax.persistence.TypedQuery;
+
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.query.Query;
+import org.hibernate.search.engine.search.query.spi.IndexSearchQuery;
+import org.hibernate.search.engine.search.query.spi.IndexSearchResult;
+import org.hibernate.search.mapper.orm.logging.impl.Log;
+import org.hibernate.search.mapper.orm.search.loading.impl.MutableObjectLoadingOptions;
+import org.hibernate.search.mapper.orm.search.query.SearchQuery;
+import org.hibernate.search.mapper.orm.search.query.SearchResult;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+
+public class HibernateOrmSearchQuery<R> implements SearchQuery<R> {
+
+	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
+
+	private final IndexSearchQuery<R> delegate;
+	private final SessionImplementor sessionImplementor;
+	private final MutableObjectLoadingOptions loadingOptions;
+
+	private HibernateOrmSearchQueryAdapter<R> wrapper;
```

Comment:
Maybe you could call `adapter` the field name or `HibernateOrmSearchQueryWrapper` the class. I would prefer the former option.